ports/cauthflow/portfile.cmake

```
@@ -0,0 +1,21 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO            offscale/cauthflow
+    REF             5d785fd8a4a47f37d13e9f65f3d3de569ed74aa4
+    SHA512          339b6b3ecdc65ba7c15899e8caae3ebae61ae031dfe2e3eb8ff8cae6c091f76cd8f9e56cf64048a85089d4a37c42ee70cb6461430af294a8447646b120a4d655
+    HEAD_REF        master
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+)
+vcpkg_cmake_install()
+file(INSTALL "${SOURCE_PATH}/cmake/LICENSE.txt"
+     DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}"
+     RENAME copyright)
+foreach(dir "lib" "share" "tools" "include")
+    if (IS_DIRECTORY "${CURRENT_PACKAGES_DIR}/debug/${dir}")
+        file(INSTALL "${CURRENT_PACKAGES_DIR}/debug/${dir}" DESTINATION "${CURRENT_PACKAGES_DIR}/${PORT}")
+    endif (IS_DIRECTORY "${CURRENT_PACKAGES_DIR}/debug/${dir}")
+endforeach (dir "lib" "share" "tools")
```

```suggestion
    endif ()
endforeach ()
```

The endforeach one isn't even accurate; I think it would be better to avoid repeating here.

---

I want to remove the hack entirely, no clue why the CMake project targets aren't being exported. https://github.com/offscale/cauthflow/blob/5d785fd/CMakeLists.txt

Any ideas how to fix this package so I can un-draft this PR? - Thanks

---

AFAICT you must add to `install(TARGETS ... EXPORT ...)`, plus `install(EXPORTS ...)`. CMake doesn't know which targets you want to export, and where they should be installed. Example: https://github.com/microsoft/vcpkg/blob/3c6e87c0b7169384acd89d82b6fd00b2cbd626ce/ports/poppler/export-unofficial-poppler.patch

---

Read again: install(TARGETS ... EXPORT ...), plus install(EXPORTS ...)

---

You mean `install(EXPORT`: https://cmake.org/cmake/help/latest/command/install.html#export

Great, got it working!